src/runtime/micro/micro_module.cc

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+*  Copyright (c) 2019 by Contributors
+* \file micro_module.cc
+*/
+
+#include <tvm/runtime/registry.h>
+#include <tvm/runtime/c_runtime_api.h>
+#include <tvm/runtime/module.h>
+#include <unordered_map>
+#include <string>
+#include "micro_session.h"
+#include "low_level_device.h"
+#include "micro_common.h"
+#include "../pack_args.h"
+
+namespace tvm {
+namespace runtime {
+/*!
+ * \brief module for uTVM micro devices
+ */
+class MicroModuleNode final : public ModuleNode {
+ public:
+  MicroModuleNode() {}
+
+  ~MicroModuleNode() {}
+
+  const char* type_key() const final {
+    return "micro";
+  }
+
+  PackedFunc GetFunction(const std::string& name,
+                         const std::shared_ptr<ModuleNode>& sptr_to_self) final;
+
+  /*!
+   * \brief initializes module by establishing device connection and loads binary
+   * \param binary_path path of the binary to be loaded
+   */
+  void InitMicroModule(const std::string binary_path) {
```

Comment:
always use const reference(const std::string&) when you don't need to modify the content